service/frontend/workflow_handler.go

```diff
@@ -884,6 +884,14 @@ func (wh *WorkflowHandler) PollWorkflowTaskQueue(ctx context.Context, request *w
 	}
 	namespaceID := namespaceEntry.ID()
 
+	// Copy WorkerVersionCapabilities.BuildId to BinaryChecksum if BinaryChecksum is missing (small
+	// optimization to save space in the poll request).
+	if request.WorkerVersionCapabilities != nil {
+		if len(request.WorkerVersionCapabilities.BuildId) > 0 && len(request.BinaryChecksum) == 0 {
+			request.BinaryChecksum = request.WorkerVersionCapabilities.BuildId
```

Comment:
I'm not following this transport optimization, why are we copying the build ID into binary checksum here?

Reply:
Wouldn't it make more sense to read VersionStamp directly in history?

Reply:
I'm just implementing this:
https://github.com/temporalio/api/blob/master/temporal/api/workflowservice/v1/request_response.proto#L247-L249

We wanted to avoid passing in a duplicate value in every poll, since binary checksum and version stamp will be the same most of the time (all the time?)

Eventually we'll get rid of binary checksum, but until then I want to just leave the downstream code alone (and the downstream code is not all in history, apparently.. look a few lines down)

Reply:
Yeah, I figured that was your intention, I thought you might want to take the next step and remove some of the references to binary checksum and use build ID directly but that can come later.